File: python/valid-mountain-array.py
class Solution:
    def validMountainArray(self, arr: List[int]) -> bool:
        peak_achieved = False
        if len(arr) < 3:
            return False
        for i in range(1, len(arr)):
            if arr[i - 1] == arr[i]:
                return False
            elif arr[i - 1] < arr[i] and peak_achieved:
                return False
            elif arr[i] < arr[i - 1]:
                if i == 1:
                    return False
                peak_achieved = True
        return peak_achieved

File: rust/valid-mountain-array.rs
impl Solution {
    pub fn valid_mountain_array(arr: Vec<i32>) -> bool {
        let mut peak_achieved = false;
        if arr.len() < 3 {
            return false;
        }
        for i in 1..arr.len() {
            if arr[i - 1] < arr[i] && peak_achieved {
                return false;
            }
            else if arr[i - 1] == arr[i] {
                return false;
            }
            else if arr[i] < arr[i - 1] {
                if i == 1 {
                    return false;
                }
                peak_achieved = true;
            }
        }
        return peak_achieved;
    }
}
